src/main/kotlin/com/dansplugins/factionsystem/dynmap/MfDynmapService.kt

```diff
@@ -51,265 +76,170 @@ class MfDynmapService(private val plugin: MedievalFactions) {
-                val lineSegmentsAtSecondPoint = lineSegments[lineSegment.second] ?: emptyList()
-                lineSegments[lineSegment.first] = lineSegmentsAtFirstPoint + lineSegment
-                lineSegments[lineSegment.second] = lineSegmentsAtSecondPoint + lineSegment
-            }
-            val isWestClaimed = claims.any { it.x == x - 1 && it.z == z }
-            if (!isWestClaimed) {
-                val lineSegment = (x to z) to (x to z + 1)
-                val lineSegmentsAtFirstPoint = lineSegments[lineSegment.first] ?: emptyList()
-                val lineSegmentsAtSecondPoint = lineSegments[lineSegment.second] ?: emptyList()
-                lineSegments[lineSegment.first] = lineSegmentsAtFirstPoint + lineSegment
-                lineSegments[lineSegment.second] = lineSegmentsAtSecondPoint + lineSegment
-            }
+    private fun updateWorldClaims(faction: MfFaction, worldId: UUID, worldClaims: List<MfClaimedChunk>, claimsMarkerSet: MarkerSet, factionInfo: String) {
+        val world = plugin.server.getWorld(worldId)
+        if (world != null) {
+            createUpdateTask(faction.id, { createAreaMarkers(faction, world, worldClaims, claimsMarkerSet, factionInfo) }) { runTaskAsynchronously(plugin) }
         }
-        if (lineSegments.isEmpty()) return emptyList()
-        val paths = mutableListOf<List<Pair<Int, Int>>>()
-        var (point, lineSegmentsAtPoint) = lineSegments.entries.first { (_, lineSegmentsAtPoint) -> lineSegmentsAtPoint.isNotEmpty() }
-        var currentPath = mutableListOf<Pair<Int, Int>>()
-        while (!lineSegments.values.all { it.isEmpty() }) {
-            currentPath.add(point)
-            val lineSegmentToFollow = lineSegmentsAtPoint.first()
-            val (first, second) = lineSegmentToFollow
-            val lineSegmentsAtFirst = lineSegments[first]
-            val lineSegmentsAtSecond = lineSegments[second]
-            if (lineSegmentsAtFirst != null) {
-                lineSegments[first] = lineSegmentsAtFirst - lineSegmentToFollow
-            }
-            if (lineSegmentsAtSecond != null) {
-                lineSegments[second] = lineSegmentsAtSecond - lineSegmentToFollow
-            }
-            point = if (first == point) second else first
-            lineSegmentsAtPoint = lineSegments[point] ?: emptyList()
-            if (!lineSegments.values.all { it.isEmpty() } && lineSegmentsAtPoint.isEmpty()) {
-                val (newPoint, newLineSegmentsAtPoint) = lineSegments.entries.first { (_, lineSegmentsAtPoint) -> lineSegmentsAtPoint.isNotEmpty() }
-                point = newPoint
-                lineSegmentsAtPoint = newLineSegmentsAtPoint
-                paths.add(currentPath)
-                currentPath = mutableListOf()
+    }
+
+    private fun createAreaMarkers(faction: MfFaction, world: World, worldClaims: List<MfClaimedChunk>, claimsMarkerSet: MarkerSet, factionInfo: String) {
+        val paths = ClaimPathBuilder.getPaths(worldClaims)
```

Comment:
Rather than instantiate a new instance of ClaimPathBuilder every time this method is called we should use a class instance

Reply:
Addressed